doc/decisions/private_api.md

```diff
@@ -0,0 +1,50 @@
+# Private API
+
+## Problem
+
+Only `libelektra-core` is supposed to access private data, but this contradicts the goal to keep the library minimal.
+`kdbprivate.h` was too generic, it contained many other parts next to the struct definitions of Key/KeySet.
+
+## Constraints
+
+## Assumptions
+
+- The [C99 standard, section 5.2.4.1](http://www.open-std.org/jtc1/sc22/wg14/) gives following limit:
+  4095 external identifiers in one translation unit
+
+## Considered Alternatives
+
+- keep current situation
+
+## Decision
+
+- Also allow other libraries (e.g. a new `libelektra-operations`) to access to non-public API.
```

Comment:
How about:

```suggestion
- Also allow other libraries (e.g. a new `libelektra-operations`) to access to non-public API.
  Such libraries need to have a good reason (e.g. performance, impossible otherwise, etc.) why they access non-public API and they need to be kept up-to-date.
  If a library cannot provide a stable API on top of the unstable non-public API, it clearly needs to state which APIs are not stable.
```

Reply:
:+1: